rest/data/project.go

```diff
@@ -259,6 +259,37 @@ func (pc *DBProjectConnector) UpdateProjectVars(projectId string, varsModel *res
 	return nil
 }
 
+func (pc *DBProjectConnector) UpdateProjectVarsByValue(toReplace, replacement string, dryRun bool) ([]*model.ProjectVars, error) {
+	catcher := grip.NewBasicCatcher()
+	matchingProjects, err := model.GetVarsByValue(toReplace)
+	if err != nil {
+		catcher.Add(errors.Wrap(err, "failed to fetch projects with matching value"))
+	}
+	if matchingProjects == nil {
+		catcher.Add(errors.New("no projects with matching value found"))
+	}
+	changes := []*model.ProjectVars{}
+	for _, project := range matchingProjects {
+		for key, val := range project.Vars {
+			if val == toReplace {
+				if !dryRun {
+					project.Vars[key] = replacement
+					_, err := project.Upsert()
+					if err != nil {
+						catcher.Add(errors.Wrapf(err, "problem overwriting variables for project '%s'", project.Id))
+					}
+				}
+				change := model.ProjectVars{
+					Id:   project.Id,
+					Vars: map[string]string{key: project.Vars[key]},
+				}
+				changes = append(changes, &change)
```

Comment:
Can we return a map of project IDs to variable key names changed? (Or something like that). I'm hesitant about returning the map of keys/values changed; for some projects, these variables might technically be private (i.e. we promise not to expose the values). It's also redundant; we know what the key is from context.

Reply:
done